datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java

```diff
@@ -159,6 +161,10 @@ public EventProducer(DatastreamTask task, TransportProvider transportProvider, C
     _enablePerTopicMetrics =
         Boolean.parseBoolean(config.getProperty(CONFIG_ENABLE_PER_TOPIC_METRICS, Boolean.TRUE.toString()));
 
+    _enablePerTopicEventLatencyMetrics =
```

Comment:
if this was written as:
```_enablePerTopicEventLatencyMetrics =         Boolean.parseBoolean(config.getProperty(CONFIG_ENABLE_PER_TOPIC_EVENT_LATENCY_METRICS,
            String.valueOf(_enablePerTopicMetrics)));``` 
you possibly won't need to check the 2 values below and can just check  if `_enablePerTopicEventLatencyMetrics` is true?

Reply:
Not sure I fully follow. If I set the default to String.valueOf(_enablePerTopicMetrics)) then I'm essentially picking up it's value (true or false), right? Which means if enablePerTopicMetrics is true and enablePerTopicEventLatencyMetrics isn't set, then enablePerTopicEventLatencyMetrics will become true too and i'll land up double counting later, right?

Reply:
Ah, you are right. I misunderstood the code.